scheduler/scheduler.cpp

```diff
@@ -723,6 +723,17 @@ static time_t prune_servers()
     }
 
     for (it = css.begin(); it != css.end();) {
+        if ((*it)->startInConnectionTest() != -1)
```

Comment:
I think we should have one call: getNextTimeout and let the cs figure out if it is a new connection or a connection timeout.